src/puppetlabs/dujour/version_check.clj

```diff
@@ -130,7 +132,7 @@
   ([request-values update-server-url callback-fn]
     (validate-config! request-values update-server-url)
     (future
-      (let [key-map (and [:certname :cacert] (keys request-values))
+      (let [key-map (vec (keys (select-keys request-values [:cacert :certname])))
```

Comment:
took me a minute to figure out what was going on here. I'm not sure how else to do this, but it would be clearer if you renamed this binding (especially since it's not a map). Maybe `keys-present` or `hashed-keys-present` that communicates what it's for.